shell/browser/electron_navigation_throttle.cc

```diff
@@ -37,10 +59,35 @@ ElectronNavigationThrottle::WillStartRequest() {
     return PROCEED;
   }
 
-  if (handle->IsRendererInitiated() && handle->IsInMainFrame() &&
-      api_contents->EmitNavigationEvent("will-navigate", handle)) {
+  std::shared_ptr<bool> is_delay_called(new bool(false));
```

Comment:
`std::shared_ptr` is banned in Chromium: https://chromium-cpp.appspot.com. We do have a few code breaking this rule but we should not add more usages.

For this check, I think implementing it in JS would be much more easy to understand, and also more efficient.

```js
      let emitted = false
      Object.defineProperty(event, 'returnValue', {
         set: (value) => {
           if (emitted) return
         },
         get: () => {}
       });
       this.emit(eventName, event, ...args);
      emitted = true
```